CafeNet/Data/Repositories/MenuItemRepository.cs

```diff
@@ -18,6 +19,71 @@ public async Task<MenuItem> CreateAsync(MenuItem item) {
             return item;
         }
 
+        public async Task<MenuItem> GetByIdAsync(long id)
+        {
+            return await _context.MenuItems.FirstOrDefaultAsync(menuItem => menuItem.Id == id);
+        }
+
+        public async Task<IEnumerable<MenuItem>> GetMenuItemsPagedAsync(int pageNumber, int pageSize)
+        {
+            return await _context.MenuItems
+                                 .Include(m => m.MenuItemVariations)
+                                 .Include(m => m.Tax)
+                                 .Skip((pageNumber - 1) * pageSize)
+                                 .Take(pageSize)
+                                 .ToListAsync();
+        }
+
+        public async Task<int> CountMenuItemsAsync()
+        {
+            return await _context.MenuItems.CountAsync();
+        }
+
+        public void DeleteById(long id)
+        {
+            var menuItem = _context.MenuItems.FirstOrDefault(u => u.Id == id);
+            if (menuItem != null)
+            {
+                _context.MenuItems.Remove(menuItem);
+                _context.SaveChanges();
+            }
+        }
+
+        public async Task<MenuItem> UpdateAvailabilityAsync(UpdateItemAvailabilityRequest request)
+        {
+            var menuItem = await _context.MenuItems
+                .Include(m => m.MenuItemVariations)
+                .Include(m => m.Tax)
+                .FirstOrDefaultAsync(m => m.Id == request.Id);
```

Comment:
Your concurrency exception does not work because on this line you get a tracked entity that updates the version field in the database. Use .AsNoTracking to avoid tracking an entity.